src/components/CustomerDetails/CollectCustomerDetailsScreen.tsx

```diff
@@ -108,14 +112,29 @@ const CollectCustomerDetailsScreen: FunctionComponent<NavigationFocusInjectedPro
   }, [shouldShowCamera]);
 
   const onCheck = async (input: string): Promise<void> => {
+    let id: string;
     try {
       setIsScanningEnabled(false);
-      const nric = validateAndCleanNric(input);
+      switch (features?.id.validation) {
+        case "NRIC":
+          id = validateAndCleanNric(input);
+          break;
+        case "REGEX":
+          const idRegex = features?.id.validationRegex;
+          id = validateAndCleanId(input, idRegex);
+          break;
+        default:
+          // Remove validation
+          id = input;
+      }
```

Comment:
Shall we move this to a generic `validateAndCleanId` utility instead?

Reply:
yep! i think we can do that, good to move this out too actually, since `AddUserModal` has the same chunk of code😛